client/lib/purchases/index.ts

```diff
@@ -220,7 +220,7 @@ export function getSubscriptionsBySite(
 					...result,
 					{
 						id: currentValue.site_id,
-						name: currentValue.site_title,
+						name: currentValue.site_title || '',
```

Comment:
I think we should also change the type of `site_title` here:

https://github.com/Automattic/wp-calypso/blob/8576def7612ece9d60a677b189ac5bc816d1f6e7/client/lib/purchases/types.ts#L348

That way any TypeScript files that rely on this data will warn us if they don't handle a `false` value.

Reply:
Well... Isn't it already in the right type (as in `string`), and never warned us? :)

Reply:
TypeScript doesn't do any validation. It's completely erased before runtime. When we add types to the data returned by an HTTP call, we are making a guarantee that the data will look the way we say it does. In this case, the types claim that `site_title` will always be a string, but the endpoint did not honor that guarantee and sometimes returned false. 

It's our job to make sure that the endpoints and the types match. If we're not sure what an endpoint might return, we should write the types as ambiguously as possible. So if we think that `site_title` might sometimes be a string, sometimes `false`, and sometimes `null` or `undefined`, we should write the type as `site_title: undefined | null | string | false`. That way the TypeScript compiler will notify us if any code that uses `site_title` tries to assume it is a string.

However, what you've done here to make sure the endpoint always returns a string simplifies things a lot because it makes the type correctly always a string.

Reply:
> TypeScript doesn't do any validation. It's completely erased before runtime.

I actually thought it was doing some validation in dev mode... But I guess it must depends on the development environment... 

> we should write the type as site_title: undefined | null | string | false. 

Yes, I definitely understand. I totally misread you this morning...
I hope there is no other place where bloginfo() is expected to be a string all the time...